imageregistry/v1/types.go

```diff
@@ -306,6 +306,59 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+type EndpointAccessibility string
+
+var (
+	InternalEndpoint = EndpointAccessibility("Internal")
+	PublicEndpoint   = EndpointAccessibility("Public")
+)
+
+type Encryption string
+
+var (
+	ClearText = Encryption("ClearText")
+	AES256    = Encryption("AES256")
+	KMS       = Encryption("KMS")
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	EncryptionType Encryption `json:"encryptionType"`
+
+	KMSEncryptionAlibaba *KMSEncryptionAlibaba `json:"kms"`
+}
+
+type KMSEncryptionAlibaba struct {
+	KeyID string `json:"keyID"`
+}
+
+// ImageRegistryConfigStorageAlibabaOSS holds Alibaba Cloud OSS configuration.
+// the registry to use Alibaba Cloud Object Storage Service for backend storage.
+// More about oss, you can look at the [official documentation](https://www.alibabacloud.com/help/product/31815.htm)
+type ImageRegistryConfigStorageAlibabaOSS struct {
+	// bucket is the bucket name in which you want to store the registry's
+	// data.
+	// Optional, will be generated if not provided.
+	// +optional
+	// About Bucket naming, more details you can look at the [official documentation](https://www.alibabacloud.com/help/doc-detail/257087.htm)
+	Bucket string `json:"bucket,omitempty"`
+	// region is the Alibaba Cloud Region in which your bucket exists.
+	// Optional, will be set based on the installed Alibaba Cloud Region.
+	// +optional
+	// For a list of regions, you can look at the [official documentation](https://www.alibabacloud.com/help/doc-detail/31837.html).
+	Region string `json:"region,omitempty"`
+	// EndpointAccessibility specifies whether the registry use the OSS VPC internal endpoint
+	// Optional, defaults to Internal.
+	// +optional
+	EndpointAccessibility EndpointAccessibility `json:"endpointAccessibility,omitempty"`
+	// encrypt specifies whether you would like your data encrypted on the server side. Defaults to false if not specified.
+	// Optional, defaults to AES256.
+	// +optional
+	// More details, you can look cat the [official documentation](https://www.alibabacloud.com/help/doc-detail/117914.htm)
+	Encrypt EncryptionAlibaba `json:"encrypt,omitempty"`
```

Comment:
omitempty + non-point struct does not work.

You probably want a pointer to the struct.